lib/utils/github/handle-issues.ts

```diff
@@ -0,0 +1,149 @@
+import {App} from '@octokit/app'
+import {Webhooks} from '@octokit/webhooks'
+import {maige} from '~/agents/maige'
+import env from '~/env.mjs'
+import {Comment} from '~/types'
+import {getRepoMeta} from '../github'
+import {incrementUsage} from '../payment'
+import {getPrompt} from '../prompt'
+
+/**
+ * Handle when a new issue is created or a comment is created on an existing issue
+ */
+export default function handleIssues(webhook: Webhooks<unknown>) {
+	webhook.on(['issues.opened', 'issue_comment.created'], async ({payload}) => {
+		const {
+			sender: {login: senderGithubUserName},
+			installation: {id: instanceId},
+			repository,
+			issue,
+			action
+		} = payload
+
+		// Handle comment being created on an issue
+		let comment: Comment = null
+		if (action === 'created') {
+			const {
+				comment: {
+					user: {login: commenterUserName},
+					body: commentBody,
+					html_url: commentHtmlUrl
+				}
+			} = payload
+
+			// Comment is made by Maige
+			if (senderGithubUserName.includes('maige'))
+				return new Response('Comment by Maige', {status: 202})
+
+			// Comment did not include reference to Maige
+			if (!payload.comment.body.toLowerCase().includes('maige'))
+				return new Response('Irrelevant comment', {status: 202})
```

Comment:
Not relevant to this PR but TODO: remove this to allow community comments to trigger runs